AsyncDisplayKit/Layout/ASDimension.h

```diff
@@ -212,6 +212,14 @@ extern ASSizeRange const ASSizeRangeZero;
 extern ASSizeRange const ASSizeRangeUnconstrained;
 
 /**
+ * Returns whether a size range has zero max width or height.
+ */
+ASDISPLAYNODE_INLINE AS_WARN_UNUSED_RESULT BOOL ASSizeRangeIsEmpty(ASSizeRange sizeRange)
+{
+  return (sizeRange.max.width == 0 || sizeRange.max.height == 0);
```

Comment:
Although these muddy our abstraction and leave us on the hook to handle invalid layout values from users in the future, the cost of being lenient here is probably justified. Let's go with it for now.

In the future, maybe we will go further and add an `ASSizeRangeNormalize` function that ensures everything is positive and than `max > min` etc.